src/pkgcheck/checks/reserved.py

```diff
@@ -101,11 +102,33 @@ def desc(self):
         return f'line {self.lineno}: {self.used_type} name "{self.line}" is disallowed because "{self.reserved_word}" is a reserved {self.reserved_type}'
 
 
+class EbuildSemiReservedName(results.LineResult, results.Warning):
+    """Ebuild uses semi-reserved variable or function name.
+
+    Ebuild is using semi-reserved variable or function names, which is likely
+    to clash with future EAPIs.
```

Comment:
I wonder if we should explicitly list the naughty variable types so people know what to avoid?